server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -277,6 +283,13 @@ public final class NodeChannels extends CloseableConnection {
 
         @Override
         public Version getVersion() {
+            // TODO: this should be the below, but in some cases the node version does not match the passed-in version.
+            // return node.getVersion();
```

Comment:
This change causes some security tests to fail - leaving it as the same as the transport version for now. This may be fixed when security code is updated to use TransportVersion